src/test/java/picard/util/IntervalListToolsTest.java

```diff
@@ -406,4 +409,63 @@ public void testLargeScatters() throws IOException {
 
         Assert.assertEquals(gather, original);
     }
+
+    @Test(timeOut = 40_000)
+    public void testCombineAbuttingIntervals() throws IOException {
+        // Test the default behavior of UNION, which is to combine abutting and overlapping intervals.
+        //gather
+        final File ilOut = File.createTempFile("IntervalListTools", ".interval_list");
+        ilOut.deleteOnExit();
+        {
+            final List<String> args = new ArrayList<>();
+            args.add("INPUT=" + abutting);
+            args.add("OUTPUT=" + ilOut);
+            args.add("ACTION=UNION"); // Note: ACTION=UNION is equivalent to UNIQUE=true and DONT_COMBINE_ABUTTING=false, or defaults
+            Assert.assertEquals(runPicardCommandLine(args), 0);
+        }
+        final IntervalList gather = IntervalList.fromFile(ilOut);
+        final IntervalList original = IntervalList.fromFile(abutting_combined);
+
+        Assert.assertEquals(gather, original); // equal to expected output
+    }
+    @Test(timeOut = 40_000)
+    public void testDontCombineAbuttingIntervals() throws IOException {
```

Comment:
Would like to see some tests of the other actions as well.  And I think you can combine this with `testCombineAbuttingIntervals` and use a DataProvider

Reply:
Fixed @kachulis